### PR TITLE
Bg_toki_hikari all functions matching

### PR DESCRIPTION
**Pull request by jpburnett**

### CODE REVIEW

**Review by louist103 — CHANGES_REQUESTED** [review]
Nice job.  Just 2 things.

File: src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c [hunk 1]
```diff
@@ -28,23 +40,152 @@ const ActorInit Bg_Toki_Hikari_InitVars = {
...
+
+void func_808BA018(BgTokiHikari* this, GlobalContext* globalCtx) {
+    s32 pad;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_toki_hikari.c", 246);
+    func_80093D18(globalCtx->state.gfxCtx);
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_toki_hikari.c", 252),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+    if (gSaveContext.linkAge == 0) {
```
Comment (fig02) [inline src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c:99]: `LINK_IS_ADULT`

File: src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c [hunk 2]
```diff
@@ -28,23 +40,152 @@ const ActorInit Bg_Toki_Hikari_InitVars = {
...
+                Actor_Kill(&this->actor);
+            }
+            break;
+    }
+}
+
+void BgTokiHikari_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void func_808B9F98(BgTokiHikari* this, GlobalContext* globalCtx) {
```
[the remark this answers is not in the document]
Comment (fig02) [reply src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c:70]: can if you want, but it prob makes sense to name with everything else in a larger docs pass. didnt read the code to see if this makes sense in context

File: src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c [hunk 3]
```diff
@@ -28,23 +40,152 @@ const ActorInit Bg_Toki_Hikari_InitVars = {
...
+    if (gSaveContext.linkAge == 0) {
+        gSPDisplayList(POLY_OPA_DISP++, D_06008190);
+    } else {
+        gSPDisplayList(POLY_OPA_DISP++, D_06007E20);
+        func_80093D84(globalCtx->state.gfxCtx);
+        gSPSegment(POLY_XLU_DISP++, 8,
+                   Gfx_TexScroll(globalCtx->state.gfxCtx, 0, globalCtx->gameplayFrames & 127, 64, 32));
+
+        gSPSegment(POLY_XLU_DISP++, 9,
+                   Gfx_TexScroll(globalCtx->state.gfxCtx, 0, globalCtx->gameplayFrames & 127, 64, 32));
```
[the remark this answers is not in the document]
Comment (jpburnett) [reply src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c:108]: The % 128 works, is that preferred over the & 127?

[the remark this answers is not in the document]
Comment (petrie911) [reply src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c:108]: Yeah. The mod is a better conceptual representation of what's happening here.

File: src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c [hunk 4]
```diff
@@ -28,23 +40,152 @@ const ActorInit Bg_Toki_Hikari_InitVars = {
...
+
+void BgTokiHikari_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void func_808B9F98(BgTokiHikari* this, GlobalContext* globalCtx) {
+}
+
+void BgTokiHikari_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgTokiHikari* this = THIS;
+    this->actionFunc(this, globalCtx);
```
Comment (petrie911) [inline src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c:75]:
```suggestion
    BgTokiHikari* this = THIS;
    
    this->actionFunc(this, globalCtx);
```

File: src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c [hunk 5]
```diff
@@ -28,23 +40,152 @@ const ActorInit Bg_Toki_Hikari_InitVars = {
...
+    } else {
+        this->unk_14C = 0.0f;
+        Actor_Kill(&this->actor);
+    }
+}
+
+void func_808BA2CC(BgTokiHikari* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_toki_hikari.c", 0x15E);
```
Comment (petrie911) [inline src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c:145]:
```suggestion
    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_toki_hikari.c", 350);
```

File: src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c [hunk 6]
```diff
@@ -28,23 +40,152 @@ const ActorInit Bg_Toki_Hikari_InitVars = {
...
+    Matrix_Translate(0.0f, 276.0f, 1122.0f, 0);
+    Matrix_Scale(0.32f, 0.32f, this->unk_14C * 7.0f, 1);
+    Matrix_RotateZ(M_PI, 1);
```
Comment (petrie911) [inline src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c:148]:
```suggestion
    Matrix_Translate(0.0f, 276.0f, 1122.0f, MTXMODE_NEW);
    Matrix_Scale(0.32f, 0.32f, this->unk_14C * 7.0f, MTXMODE_APPLY);
    Matrix_RotateZ(M_PI, MTXMODE_APPLY);
```

File: src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c [hunk 7]
```diff
@@ -28,23 +40,152 @@ const ActorInit Bg_Toki_Hikari_InitVars = {
...
+    func_80093D18(globalCtx->state.gfxCtx);
+    Matrix_Push();
+    gDPPipeSync(POLY_XLU_DISP++);
+    gDPSetPrimColor(POLY_XLU_DISP++, 0, 0x80, (this->unk_14C * 255.0f), ((u8)(155.0f * this->unk_14C) + 0x64),
+                    (this->unk_14C * 255.0f), (this->unk_14C * 255.0f));
+
+    gDPSetEnvColor(POLY_XLU_DISP++, ((u8)(this->unk_14C * 155.0f) + 0x64), (u8)(255.0f * this->unk_14C), 0, 128);
```
Comment (petrie911) [inline src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c:155]:
```suggestion
    gDPSetPrimColor(POLY_XLU_DISP++, 0, 0x80, this->unk_14C * 255.0f, (u8)(155.0f * this->unk_14C) + 100,
                    this->unk_14C * 255.0f, this->unk_14C * 255.0f);

    gDPSetEnvColor(POLY_XLU_DISP++, (u8)(this->unk_14C * 155.0f) + 100, this->unk_14C * 255.0f, 0, 128);
```
Makes more sense in decimal, and the parentheses are unneeded. Can SetEnvColor be written like this and still match? If not, does it work to have `(u8)` casts on all the other products?

File: src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c [hunk 8]
```diff
@@ -28,23 +40,152 @@ const ActorInit Bg_Toki_Hikari_InitVars = {
...
+
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_toki_hikari.c", 0x17E),
```
Comment (petrie911) [inline src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c:157]:
```suggestion
    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_toki_hikari.c", 382),
```

**Review by petrie911 — COMMENTED** [review]
Just cleanup stuff.

File: src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c (hunk 7 above)
Comment (jpburnett) [reply src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c:155]: It still needs to have the (u8) cast to work even with the change :/

Comment (fig02) [reply src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c:155]: what makes it not match anymore? flipping the order of the adds?

Comment (jpburnett) [reply src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c:155]: Having the `(u8)(255.0f * this->unk_14C)` will match, if I try the suggested `this->unk_14C * 255.0f` it causes the mismatch. I tried changing the other instances of having `this->unk_14C * <float_num>` too to see if that was the issue, but it did not end up matching.

Comment (fig02) [reply src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c:155]: so just flipping the mult causes the non match, even if you leave the cast there?

**Review by fig02 — APPROVED** [review]

**Review by Roman971 — APPROVED** [review]

**Pull request merged**